shell/packages/sandstorm-ui-powerbox/powerbox-server.js

```diff
@@ -41,28 +44,61 @@ Meteor.methods({
 
     const db = this.connection.sandstormDb;
 
+    const session = db.collections.sessions.findOne(sessionId);
+    if (!session) {
+      throw new Meteor.Error(403, "Invalid session ID");
+    }
+
     let descriptor;
+    let requirements = [];
     if (frontendRefVariety.ipNetwork) {
       if (!db.isAdmin(this.userId)) {
         throw new Meteor.Error(403, "User must be an admin to powerbox offer IpNetwork");
       }
 
       descriptor = encodePowerboxDescriptor({ tags: [{ id: Ip.IpNetwork.typeId }] });
+      requirements.push({ userIsAdmin: Meteor.userId() });
     } else if (frontendRefVariety.ipInterface) {
       if (!db.isAdmin(this.userId)) {
         throw new Meteor.Error(403, "User must be an admin to powerbox offer IpInterface");
       }
 
       descriptor = encodePowerboxDescriptor({ tags: [{ id: Ip.IpInterface.typeId }] });
+      requirements.push({ userIsAdmin: Meteor.userId() });
```

Comment:
This requirement is not new. If it's wrong, we should remove it in a separate change. (But I'd argue, as usual, that what we really want is some sort of UI where when an admin is revoked, the revoker can audit what connections they made and choose whether or not to revoke them.)